### DISCUSSION TITLE
Has anyone tried creating more than one table of content (TOC)?

### DISCUSSION THREAD

**Discussion by DRomeroLucas** [mpdf/mpdf#1955] [Q&A]
Hello everyone.

I'm in a project using this great library, I'm trying to create more than one TOC cause I need to represent the general index, the index of figures and an index of tables, for the first TOC I was using the _<tocpagebreak/>_, but when I want to create a new TOC using another _<tocpagebreak/>_ this doesn't work, I mean, I put into the tag _name_ for specify the TOC, but when I put this setting in the tags or even in just one of them the TOC content disappear, I was reading the documentation but don't specify the way to use the _$name_ variable, I was assumming this can use like the others properties deleting the dolar sign ($).

So I want to know if someone already create a document with more than one TOC and if is possible that you can explain us how is possible create more than one of this.

**Comment by jakejackson1** [comment]
I couldn't find any specific documentation about it, but I know this test produces a PDF with multiple TOCs: https://github.com/mpdf/mpdf/blob/d748f1e90eb98a3f61d8a5ae8035ba1972ef268c/tests/Mpdf/TocNumbering.php#L76-L128

That should help point you in the right direction.

**Reply by DRomeroLucas** [reply]
Hi Jake.

Interesting, ok well, It seems the functionality that I expected doesn't exist, but ist's ok, maybe later I'll try to create a PHP function to create that tables, the piece of code that you share there is truly helpful, I really apreciate it, thank you so much Jake.

**Comment by juuce** [comment]
Hi, Sorry to hijack this thread but I too am having issues with multiple TOC in a document. There appears to be a few issues with multiple TOC in a document.
1. When adding a second TOC multiple warnings appear 
`
Notice: Uninitialized string offset: 0 in D:\www\vhosts\vendor\mpdf\mpdf\src\Mpdf.php on line 16724
Notice: Uninitialized string offset: 0 in D:\www\vhosts\vendor\mpdf\mpdf\src\Mpdf.php on line 16726`

mpdf v 8.2.3

This is due to $b['c'] being a string instead of an array. hacking the function to check if is_array() generates the PDF with the 2 TOC BUT...
Question: Why does a TOC always need to have a page break? Is there a method to put 2 TOC on the same page?

Is this possible? (See image below) - 4 different TOC sections.
![image](https://github.com/mpdf/mpdf/assets/8197867/4fb4a000-da69-4fd0-ade3-05a69f3154c0)

Thanks.
